File: array/python/element_appearing_ore_than_25_in_sorted_array.py
from typing import List
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        n = len(arr)
        c,k = arr[0],0

        for i in range(0,n):
            if arr[i] == c:
                k+=1
                if k*4 > n: return c
            else:
                c,k = arr[i],1

        return c

File: array/ts/element_appearing_ore_than_25_in_sorted_array.ts
function findSpecialInteger(arr: number[]): number {
    const n: number= arr.length;
    let c = arr[0];
    let k = 0;

    for (let i = 0;i < n;i++) {
        if (arr[i] === c) {
            k+=1
            if (k*4 > n ){
                return c
            }
        }else{
            c = arr[i]
            k = 1
        }
    }
    return c
}